### PR TITLE
Addaptive status box

### PR DESCRIPTION
**Pull request by rmosimann**
This is a ShortOne. Added a method to Change the StatusDisplay in the Request and Support view.

### CODE REVIEW

**Review by martinfrancois — APPROVED** [review]
Thanks for the PR! Looks good to me, I just noticed it doesn't update when I click the refresh button, do you know why?

File: src/ch/imedias/rsccfx/model/Rscc.java [hunk 1]
```diff
@@ -166,24 +171,55 @@ public void killConnection() {
...
+   *
+   * @param text             Text to show for the connection status.
+   * @param statusStyleIndex Index of the connectionStatusStyles.
+   */
+  public void setConnectionStatus(String text, int statusStyleIndex) {
+    if (statusStyleIndex < 0 || statusStyleIndex >= connectionStatusStyles.length || text == null) {
+      throw new IllegalArgumentException();
+    }
+    setConnectionStatusText(text);
+    setConnectionStatusStyle(getConnectionStatusStyles(statusStyleIndex));
```
Comment (l-marchesi) [inline src/ch/imedias/rsccfx/model/Rscc.java:214]: :+1: I like how you implemented this feature!

File: src/ch/imedias/rsccfx/view/RsccRequestPresenter.java [hunk 2]
```diff
@@ -80,6 +81,20 @@ private void attachEvents() {
     view.predefinedAddressesPane.setOnMouseClicked(
         event -> view.keyGeneratorPane.setExpanded(false)
     );
+
+    model.connectionStatusStyleProperty().addListener((observable, oldValue, newValue) -> {
+      Platform.runLater(() -> {
+        view.statusBox.getStyleClass().clear();
+        view.statusBox.getStyleClass().add(newValue);
```
Comment (l-marchesi) [inline src/ch/imedias/rsccfx/view/RsccRequestPresenter.java:92]: I like this as well :-)

**Review by l-marchesi — APPROVED** [review]
Great, merge it :+1:

**Pull request merged**